lib/archivers/zip/zip-archive-output-stream.js

```diff
@@ -88,8 +88,10 @@ ZipArchiveOutputStream.prototype._appendBuffer = function(ae, source, callback)
 };
 
 ZipArchiveOutputStream.prototype._appendStream = function(ae, source, callback) {
-  ae.getGeneralPurposeBit().useDataDescriptor(true);
-  ae.setVersionNeededToExtract(constants.MIN_VERSION_DATA_DESCRIPTOR);
+  if (ae.getMethod() === constants.METHOD_DEFLATED) {
```

Comment:
i feel like this is going to cause issues. because with streaming we dont have the size yet. so the code in _writeLocalFileHeader (https://github.com/archiverjs/node-compress-commons/blob/master/lib/archivers/zip/zip-archive-output-stream.js#L405) is going to write zeros for size. since normally when a data descriptor is used the size is fixed at the end.

to keep the streaming interface and use this logic, we would have to buffer all the content leading to higher memory usage.

Reply:
I was following similar logic [here](https://github.com/archiverjs/node-compress-commons/blob/master/lib/archivers/zip/zip-archive-output-stream.js#L149). But I've since updated the PR.